hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/OptionsResolver.java

```diff
@@ -190,7 +202,18 @@ public static boolean needsAsyncClustering(Configuration conf) {
    * @param conf The flink configuration.
    */
   public static boolean needsScheduleClustering(Configuration conf) {
-    return isInsertOperation(conf) && conf.getBoolean(FlinkOptions.CLUSTERING_SCHEDULE_ENABLED);
+    if (!conf.getBoolean(FlinkOptions.CLUSTERING_SCHEDULE_ENABLED)) {
+      return false;
+    }
+    WriteOperationType operationType = WriteOperationType.fromValue(conf.getString(FlinkOptions.OPERATION));
+    if (OptionsResolver.isConsistentHashingBucketIndexType(conf)) {
+      // Write pipelines for table with consistent bucket index would detect whether clustering service occurs,
+      // and automatically adjust the partitioner and write function if clustering service happens.
+      // So it could handle UPSERT and INSERT case.
```

Comment:
What the behavior of UPSERT + consistent hashing index ? For insert operation, we do not expect to deduplicate, while the log file readers always merge the record payloads.

Reply:
Update the logical of `needsScheduleClustering` because the previous version only support `insert` for schedule clustering. I think the reason to add this limitation is the mapping from key to file group is changing after the clustering . But for consistent bucket index, both partitioner and write function in the write pipelines could handle the clustering happens.

Reply:
Scheduling for resize plan only apply for upsert write operation type instead of insert.
Because insert would not generate file group id based on index.